var/spack/repos/builtin/packages/py-isort/package.py

```diff
@@ -16,13 +16,23 @@ class PyIsort(PythonPackage):
     version("5.10.1", sha256="e8443a5e7a020e9d7f97f1d7d9cd17c88bcb3bc7e218bf9cf5095fe550be2951")
     version("5.9.3", sha256="9c2ea1e62d871267b78307fe511c0838ba0da28698c5732d54e2790bf3ba9899")
     version("5.9.1", sha256="83510593e07e433b77bd5bff0f6f607dbafa06d1a89022616f02d8b699cfcd56")
-    version("4.3.20", sha256="c40744b6bc5162bbb39c1257fe298b7a393861d50978b565f3ccd9cb9de0182a")
-    version("4.2.15", sha256="79f46172d3a4e2e53e7016e663cc7a8b538bec525c36675fcfd2767df30b3983")
+    version(
+        "4.3.20",
+        sha256="c40744b6bc5162bbb39c1257fe298b7a393861d50978b565f3ccd9cb9de0182a",
+        deprecated=True,
+    )
+    version(
+        "4.2.15",
+        sha256="79f46172d3a4e2e53e7016e663cc7a8b538bec525c36675fcfd2767df30b3983",
+        deprecated=True,
+    )
 
     variant("colors", default=False, description="Install colorama for --color support")
 
     depends_on("python@3.8:", when="@5.12:", type=("build", "run"))
-    depends_on("python@3.6.1:3", when="@5:5.10", type=("build", "run"))
+    # This needs to be @3.6 since for bootstrapping the current Spack interpreter is
+    # identified by major.minor (and the new versioning identifies it as @=3.6)
+    depends_on("python@3.6:", when="@5:", type=("build", "run"))
```

Comment:
To prevent someone getting `isort@5` if they have the (presumably) broken Python 3.6.0, would it be reasonable to add:
```python
    conflicts("python@3.6.0", when="@5:5.10")
```
?

Reply:
Yes, I didn't add the clause since I thought this is very unlikely to happen - but what you say is correct.  @adamjstewart Do you have preferences? I'm fine with both choices (having the clause or not).

Reply:
@greenc-FNAL Is this working for you? I tested on a Docker image, wondering if the fix also works on a "real" system :slightly_smiling_face:

Reply:
Looks like everything is fine from the point of view of this PR, thank you.

On a related note though, it would be nice to have a `spack bootstrap status --show-installed` (or even `spack find --bootstrap`) or similar, so one can tell exactly what packages have been installed for bootstrapping purposes.

Reply:
@greenc-FNAL 
```console
$ spack -b find
-- linux-centos7-x86_64 / gcc@10.2.1 ----------------------------
bison@3.0.4  clingo-bootstrap@spack  clingo-bootstrap@spack  cmake@3.27.6  patchelf@0.18.0  python@3.8.18  python@3.11.5

-- linux-ubuntu20.04-x86_64 / gcc@9.4.0 -------------------------
git@2.25.1           py-editables@0.3                   py-hatch-vcs@0.3.0   py-mccabe@0.7.0           py-pathspec@0.11.1      py-poetry-core@1.6.1   py-setuptools@68.0.0           py-types-psutil@5.9.5.16      py-wheel@0.41.2
py-black@23.1.0      py-flake8@6.1.0                    py-hatchling@1.18.0  py-mypy@1.3.0             py-pip@23.1.2           py-pycodestyle@2.11.0  py-setuptools-scm@7.1.0        py-types-setuptools@68.2.0.0  python@3.11
py-calver@2022.6.26  py-flit-core@3.9.0                 py-iniconfig@2.0.0   py-mypy-extensions@1.0.0  py-platformdirs@3.10.0  py-pyflakes@3.1.0      py-tomli@2.0.1                 py-types-typed-ast@1.5.8.7
py-click@8.1.7       py-hatch-fancy-pypi-readme@23.1.0  py-isort@5.12.0      py-packaging@23.1         py-pluggy@1.0.0         py-pytest@7.3.2        py-trove-classifiers@2023.8.7  py-typing-extensions@4.8.0
==> 41 installed packages
```

Reply:
Sorry, missed that when I looked! My excuse is Friday.

Reply:
No worries, we have plenty of commands / options It's easy to miss one! :wink:

Reply:
I would add a conflict, or just drop Python 3.6 support for bootstrapping style deps. Also needs an upper bound on Python version.